org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceDiagnosticsHandler.java

```diff
@@ -128,6 +143,66 @@ else if (projectsManager.isBuildFile(file)) {
 		return false;
 	}
 
+	public List<IMarker> publichDiagnostic(IProgressMonitor monitor) throws CoreException {
+		List<IMarker> problemMarkers = getProblemMarkers(monitor);
+		publishDiagnostics(problemMarkers);
+		return problemMarkers;
+	}
+
+	private List<IMarker> getProblemMarkers(IProgressMonitor monitor) throws CoreException {
+		IProject[] projects = ResourcesPlugin.getWorkspace().getRoot().getProjects();
+		List<IMarker> markers = new ArrayList<>();
+		for (IProject project : projects) {
+			if (monitor != null && monitor.isCanceled()) {
+				throw new OperationCanceledException();
+			}
```

Comment:
ignore default project, I think

Reply:
It turns out this is the problem markers, but it will be ignored by publish diagnostics, which is the same behavior as the current design.